### DISCUSSION TITLE
Multiple Handler Support

### DISCUSSION THREAD

**Discussion by tybruno** [carlmontanari/scrapli#139] [Ideas]
Hello,

One of the tools I am building with Scrapli involves sending multiple commands to multiple devices. There are a few futures my tool is needing. I am going to create these features but I think they could also be a part of the Scrapli core library.

### GenerateHandlers
Say I have 10+ devices that I want to connect to. They all use the same auth but they have different hosts. It would be awesome to have a generic generator that can help generate the handlers for each host. `GeneratHandlers(hosts:iterable[str]) -> Generator`

### Runner
After we have created the handlers we now will want to provide the commands we want to send through each of these handlers.

I think features would provide the users of Scrapli with being able to more easily work with multiple devices.  

Do you think these features would be worth adding?

**Comment by carlmontanari** [comment]
👋 

Probably not something I want to add to scrapli. Its obviously a good/useful thing, but I want to do as little as possible in scrapli basically, and I think this would be a slippery slope for feature bloat!

**Reply by tybruno** [reply]
That makes sense. I'll add it to my abstraction.
